### PR TITLE
A few heap related tweaks

### PR DESCRIPTION
**Pull request by tannewt**
* Long live byte strings
* Long live code.py code
* Collect after import
* Limit max QSTR pool size

### CODE REVIEW

**Comment by tannewt** [comment]
Travis is testing nrf fix.

In the meantime I forgot to post the videos:

before: https://www.youtube.com/watch?v=dabIOpBfnks
after: https://www.youtube.com/watch?v=Q_hXDPc1L8Q

File: ports/nrf/Makefile [hunk 1]
```diff
@@ -213,7 +213,7 @@ FROZEN_MPY_PY_FILES := $(shell find -L $(FROZEN_MPY_DIR) -type f -name '*.py')
 FROZEN_MPY_MPY_FILES := $(addprefix $(BUILD)/,$(FROZEN_MPY_PY_FILES:.py=.mpy))
 
 OBJ += $(PY_O) $(SUPERVISOR_O) $(addprefix $(BUILD)/, $(SRC_C:.c=.o))
-OBJ += $(BUILD)/pins_gen.o
+OBJ += $(BUILD)/pins_gen.o supervisor/cpu.o
```
Comment (arturo182) [inline ports/nrf/Makefile]: Adding a SRC_S variable and then using addprefix to change to .o would probably be cleaner :)

Comment (tannewt) [reply ports/nrf/Makefile]: Done. Thanks for the comment

File: ports/atmel-samd/Makefile [hunk 2]
```diff
@@ -412,13 +412,16 @@ endif
 SRC_SHARED_MODULE_EXPANDED = $(addprefix shared-bindings/, $(SRC_SHARED_MODULE)) \
                              $(addprefix shared-module/, $(SRC_SHARED_MODULE))
 
+SRC_O = supervisor/$(CHIP_FAMILY)_cpu.o
+
```
Comment (dhalbert) [inline ports/atmel-samd/Makefile]: Same thing here that you did in the nrf makefile?

Comment (tannewt) [reply ports/atmel-samd/Makefile]: Done! Please take another look.

**Comment by tannewt** [comment]
@dhalbert Please take a look!

**Pull request merged**